docs/designs/ps_design.md

```diff
@@ -1,53 +1,62 @@
 # Parameter Server Design
 
-
 ## Overview
-Currently, there is one parameter server (PS) co-existed with the master. In order to support PS with multiple nodes and PS fault tolerance, we need to separate PS from the master. Besides a KV store for model parameters, PS should also support updating parameters using gradients. Currently, models have two kinds of parameter, variable and embedding vector. PS should support model variable update and embedding table sparse update.
 
-The master will create a PS with *N* PS nodes, where *N* is specified by the user. Each model variable and embedding vector has a corresponding PS node. Thus, every PS node has a subset of model variables and embedding tables.
+A typical parameter server architecture of data parallelism distributed training system contains two roles:
 
-The master will monitor PS nodes status similar to what it does for workers. In case a PS node fails, the master will try to relaunch it. Since PS nodes have higher priority than workers, if there are still some running workers, the relaunch will succeed by using either idle or preempted Kubernetes resources. If the relaunch fails, there are no workers left. The Elastic job has to wait for resources for the PS node and workers.
+- parameter server(or PS for short), maintaining the global model, and providing parameter pull/push/optimize/checkpoint service
+- worker, computing gradients of model parameters using local data and pushing gradients to parameter servers.
 
-Each worker has a local copy of the model variables. After the master relaunches a PS node, the PS node can recover model variables from workers. For embedding vectors, PS must create replicas to support fault tolerance. For each PS node *PS(i)*, it will store *M* replicas in the following *M* PS nodes from *PS(i+1 % N)* to *PS(i+M % N)*. The relaunched PS node can recover embedding vectors from one of its replicas. If there are more than *M* continuously-indexed PS nodes failing, at least one PS node fails with all of its replicas. The ElasticDL job has to recover from a recent checkpoint.
+Sharding the model to multiple parameter servers is a common choice. The reasons are as following:
```

Comment:
Again, the enumeration here prevents us from explaining the ideas progressively. Please be aware that these ideas are not on the same logic level.  A single-node parameter server architecture is a bottleneck of communication and computation anyway, in spite of the model size.  So it would be easier to say:

> Many systems use multiple parameter server instances in a job, especially when the model could be large and overruns the memory space of a process. In such cases, we can shard the model and distribute the pieces to multiple parameter server instances. Even if the model is not too big, model sharding prevents the case that a single-node parameter server architecture becomes the bottleneck of communication and computation.